examples/ensemble/plot_stack_predictors.py

```diff
@@ -244,27 +242,38 @@ def plot_regression_results(ax, y_true, y_pred, title, scores, elapsed_time):
 fig, axs = plt.subplots(2, 2, figsize=(9, 7))
 axs = np.ravel(axs)
 
+splits = namedtuple("splits", ("train", "test"))
+cv = [splits(*data) for data in KFold(n_splits=5).split(X, y)]
+
 for ax, (name, est) in zip(
     axs, estimators + [("Stacking Regressor", stacking_regressor)]
 ):
-    start_time = time.time()
-    score = cross_validate(
-        est, X, y, scoring=["r2", "neg_mean_absolute_error"], n_jobs=-1, verbose=0
+    start_time = time.perf_counter()
+    cv_results = cross_validate(
+        est,
+        X,
+        y,
+        cv=cv,
+        scoring=["r2", "neg_mean_absolute_error"],
+        n_jobs=-1,
+        verbose=0,
+        return_estimator=True,
     )
-    elapsed_time = time.time() - start_time
-
-    y_pred = cross_val_predict(est, X, y, n_jobs=-1, verbose=0)
+    elapsed_time = time.perf_counter() - start_time
+    y_pred = np.zeros_like(y)
+    for estimator, fold in zip(cv_results["estimator"], cv):
```

Comment:
With my previous comment, you can get the training and testing indices using `cv.split`.